### PR TITLE
Add XT65 keyboard

### PR DESCRIPTION
**Pull request by kopibeng**
## Description

Adds QMK support for XT65 keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Adds QMK support for XT65 keyboard.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kopibeng/xt65/config.h [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4B50 // 'KP' kopibeng  
+#define PRODUCT_ID      0x0650 
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    kopibeng
+#define PRODUCT         XT65
+#define DESCRIPTION     kopibeng XT65
```
Comment (drashna) [inline keyboards/kopibeng/xt65/config.h:27]:
```suggestion
```

File: keyboards/kopibeng/xt65/readme.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#-- XT65 --#
```
Comment (drashna) [inline keyboards/kopibeng/xt65/readme.md:1]:
```suggestion
# XT65
```

And could you add instructions on how to enter the bootloader to reflash the board?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/kopibeng/xt65/readme.md (hunk 2 above)
Comment (kopibeng) [reply keyboards/kopibeng/xt65/readme.md]: All done, also added image of supported layout

File: keyboards/kopibeng/xt65/rules.mk [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/kopibeng/xt65/rules.mk:11]:
```suggestion
```

File: keyboards/kopibeng/xt65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (drashna) [inline keyboards/kopibeng/xt65/rules.mk:32]:
```suggestion
AUDIO_ENABLE = no           # Audio output
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by kopibeng** [comment]
Hello @drashna, could I get some help in getting another reviewer for this PR to get it merged please?

File: keyboards/kopibeng/xt65/config.h [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+//#define MAGIC_KEY_LAYER8         8
+//#define MAGIC_KEY_LAYER9         9
+//#define MAGIC_KEY_BOOTLOADER     B
+//#define MAGIC_KEY_BOOTLOADER_ALT ESC
+//#define MAGIC_KEY_LOCK           CAPS
+//#define MAGIC_KEY_EEPROM         E
+//#define MAGIC_KEY_EEPROM_CLEAR   BSPACE
+//#define MAGIC_KEY_NKRO           N
+//#define MAGIC_KEY_SLEEP_LED      Z
+
```
Comment (fauxpark) [inline keyboards/kopibeng/xt65/config.h:136]:
```suggestion
```

File: keyboards/kopibeng/xt65/readme.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+![XT65](https://i.imgur.com/extkBr1.png)
+
+A 65% PCB with support for VIA, LED backlight and RGB underglow.
+
+Keyboard Maintainer: kopibeng
+
+Hardware Supported: A 65% keyboard with ATMEGA32U4
+
+Hardware Availability: N/A
```
Comment (fauxpark) [inline keyboards/kopibeng/xt65/readme.md:11]:
```suggestion
* Keyboard Maintainer: kopibeng
* Hardware Supported: A 65% keyboard with ATMEGA32U4
* Hardware Availability: N/A
```

File: keyboards/kopibeng/xt65/readme.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+	Press switch (SW1) on bottom side of PCB to enter bootloader
+	make kopibeng/xt65:default
```
Comment (fauxpark) [inline keyboards/kopibeng/xt65/readme.md:16]:
```suggestion
    make kopibeng/xt65:default
    
Press the switch (SW1) on bottom side of PCB to enter bootloader.
```

File: keyboards/kopibeng/xt65/rules.mk [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+
+LAYOUTS = 65_7u_split_bs 65_625u
```
Comment (fauxpark) [inline keyboards/kopibeng/xt65/rules.mk:25]:
```suggestion
```

File: keyboards/kopibeng/xt65/rules.mk [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes      # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
```
Comment (fauxpark) [inline keyboards/kopibeng/xt65/rules.mk:21]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**